lib/ansible/executor/playbook_executor.py

```diff
@@ -186,7 +186,7 @@ def run(self):
                             if C.RETRY_FILES_SAVE_PATH:
                                 basedir = C.shell_expand(C.RETRY_FILES_SAVE_PATH)
                             elif playbook_path:
-                                basedir = os.path.dirname(playbook_path)
+                                basedir = os.path.dirname(os.path.realpath(playbook_path))
```

Comment:
Hm.  If the playbook is a symlink (playbooks/mybook.yml -> ../elsewhere/mybook.yml), then this will create the retry files in elsewhere/ rather than in playbooks/.  Is that desirable?